kimchi/src/circuits/constraints.rs

```diff
@@ -185,6 +155,45 @@ pub struct ConstraintSystem<F: FftField> {
     /// lookup constraint system
     #[serde(bound = "LookupConstraintSystem<F>: Serialize + DeserializeOwned")]
     pub lookup_constraint_system: Option<LookupConstraintSystem<F>>,
+
+    /// precomputes
+    #[serde(skip)]
+    pub precomputations: Arc<DomainConstantEvaluations<F>>,
+}
+
+impl<F: FftField> Default for ConstraintSystem<F> {
```

Comment:
why implement `Default` for `ConstraintSystem`?

Reply:
ok I see that you need it because of serde skip. So two things:

* in another comment I say that we shouldn't skip this, so we actually don't need Default here
* you could have done `#[derive(Default)]` and avoid that code :)